patches/server/0727-Fix-ServerLevel-Iteration-CME.patch

```diff
@@ -0,0 +1,30 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: thamid-gamer <60953955+thamid-gamer@users.noreply.github.com>
+Date: Tue, 13 Jul 2021 13:53:13 -0400
+Subject: [PATCH] Fix ServerLevel Iteration CME
+
+
+diff --git a/src/main/java/net/minecraft/server/MinecraftServer.java b/src/main/java/net/minecraft/server/MinecraftServer.java
+index 3dded5c491ace6b073a7bc3178976bd70f0b9393..bf1d0c6670d7b3de68c553b6782016e141a14c71 100644
+--- a/src/main/java/net/minecraft/server/MinecraftServer.java
++++ b/src/main/java/net/minecraft/server/MinecraftServer.java
+@@ -1496,7 +1496,7 @@ public abstract class MinecraftServer extends ReentrantBlockableEventLoop<TickTa
+         MinecraftTimings.commandFunctionsTimer.stopTiming(); // Spigot // Paper
+         midTickLoadChunks(); // Paper
+         this.profiler.popPush("levels");
+-        Iterator iterator = this.getAllLevels().iterator();
++        // Iterator iterator = this.getAllLevels().iterator(); // Paper - unnecessary
+ 
+         // CraftBukkit start
+         // Run tasks that are waiting on processing
+@@ -1528,8 +1528,8 @@ public abstract class MinecraftServer extends ReentrantBlockableEventLoop<TickTa
+         // Paper end
+         MinecraftTimings.timeUpdateTimer.stopTiming(); // Spigot // Paper
+ 
+-        while (iterator.hasNext()) {
+-            ServerLevel worldserver = (ServerLevel) iterator.next();
++        for (ServerLevel worldserver : this.levels.values().toArray(new ServerLevel[0])) { // Paper - avoid CME
```

Comment:
I'd be much more inclined to simply tell people to delay their world creation than doing this every single tick... don't know if this is at all something we should consider.

Is it possible to rather make `this.levels` a copy-on-write collection instead to avoid the issue?